### ISSUE TITLE
'DateDefaultTimezone' not working as expected

### ISSUE THREAD

**Issue by ecdarreola** [rathena/FluxCP#285]
Not sure if I understand the 'DateDefaultTimezone' properly but I'll try.

So I configured it to be 
`'DateDefaultTimezone'		=> 'Asia/Singapore',`

Resulting to any logs that will be sent to the database will be in Singapore time or (GMT+8) or (UTC+8).

However, there's a problem with confirm expire wherein the scenario is.
**Web Server Local Time (Host)** is using UTC (+00:00) and FluxCP Timezone is above which is **Asia/Singapore**

E.G. of problem
A user created an account with an expiration of 24 hours (email).
Details:
- Time of Account Creation is 2021-05-01 08:05:03 (reg_date)
- Confirm expire will be 2021-05-02 08:05:03 (24 hours).

Now user forgot to confirm email upon registration. Luckily confirmation account was successfully done the next day at 8:06 AM of May 02 which is 1 minute after the confirm expiry.

What I've found out is that:
- DateDefaultTimezone works on logs or in creating values.
- FluxCP uses local time of **website host (UTC)** and not the default timezone.

So basically, if hosting/local time is in UTC and FluxCP is at Asia/Singapore or (UTC+8) the confirm expiry will be 24 + 8 hours or 32 hours.

**Comment by Akkarinage** [comment]
You could also throw into the mix the timezone of your database server, as some queries use the `NOW()` function instead of parsing the timestamp from the webserver.

It should work as intended though, as all timestamps generated by FluxCP use the timezone specified.

It is gathered:
https://github.com/rathena/FluxCP/blob/3a96b24c455954924139774a76e17156be1233d2/index.php#L74

And set:
https://github.com/rathena/FluxCP/blob/3a96b24c455954924139774a76e17156be1233d2/index.php#L75

**Comment by ecdarreola** [comment]
@Akkarinage Apologies if I can't follow correctly, so the 24 + 8 the difference between UTC and Singapore is valid? How do you push the webserver to use only the ```DateDefaultTimezone```

**Comment by ecdarreola** [comment]
As I've checked the ```account\confirm.php``` it uses the NOW()
```$sql .= "userid = ? AND confirm_code = ? AND confirmed = 0 AND confirm_expire > NOW() LIMIT 1";```

So by default, if the webserver local timezone is in UTC 00:00 or -04:00, the confirm_expire at Asia/Singapore zone will add the hours offset?

**Comment by ecdarreola** [comment]
> You could also throw into the mix the timezone of your database server, as some queries use the `NOW()` function instead of parsing the timestamp from the webserver.
> 
> It should work as intended though, as all timestamps generated by FluxCP use the timezone specified.
> 
> It is gathered:
> https://github.com/rathena/FluxCP/blob/3a96b24c455954924139774a76e17156be1233d2/index.php#L74
> 
> And set:
> https://github.com/rathena/FluxCP/blob/3a96b24c455954924139774a76e17156be1233d2/index.php#L75

Closing the issue now. Thanks mate!

In the event someone else also experience this. Make sure that the DB is in sync with the time of your local host and use NOW() in all SQL query.
